src/rgw/rgw_sal_motr.cc

```diff
@@ -1511,26 +1509,24 @@ void MotrObject::read_bucket_info(const DoutPrefixProvider* dpp, std::string& bn
 }
 
 int MotrObject::modify_obj_attrs(RGWObjectCtx* rctx, const char* attr_name, bufferlist& attr_val, optional_yield y, const DoutPrefixProvider* dpp)
-{
+{  
   rgw_obj target = get_obj();
-  int r = get_obj_attrs(rctx, y, dpp, &target);
-  if (r < 0) {
-    return r;
-  }
+  sal::Attrs set_attrs;
+
   set_atomic(rctx);
-  attrs[attr_name] = attr_val;
```

Comment:
Even though these renaming doesnt affect functionally, using proper references helps in readability & using for their intended use atleast. If these are the only reasons, then am ok

Reply:
Earlier it was directly making changes in attrs, but now set_attrs is a new sal::Attrs type map which will be passes to set_obj_attrs method.